homeassistant/components/duotecno/entity.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, unit: BaseUnit) -> None:
             manufacturer="Duotecno",
             name=unit.get_node_name(),
         )
-        self._attr_unique_id = f"{unit.get_node_address()}-{unit.get_number()}"
+        self._attr_unique_id = f"{self.entity_id}-{unit.get_node_address()}-{unit.get_number()}"
```

Comment:
- I think changing the unique ID is a breaking change.
-  we normally recommend using a migration (see `__init__.py` changes in #142804), and have proper associated tests (see `test_init.py` changes in #142804)
- I don't think `self.entity_id` is really unique - if the system does not provide a proper unique ID (eg. a serial number) then the entry_id of the config entry should be used.